File: algorithms/1748.sum-of-unique-elements/solution.go
func sumOfUnique(nums []int) int {
	sum := 0
	m := map[int]int{}

	for _, n := range nums {
		if _, ok := m[n]; ok {
			m[n] += 1
		} else {
			m[n] = 1
		}
	}

	for k, v := range m {
		if v == 1 {
			sum += k
		}
	}

	return sum
}

File: algorithms/1748.sum-of-unique-elements/solution.rs
impl Solution {
    pub fn sum_of_unique(nums: Vec<i32>) -> i32 {
        use std::collections::HashMap;

        let mut sum: i32 = 0;
        let mut map: HashMap<i32, i32> = HashMap::new();

        for n in nums {
            let value = map.entry(n).or_insert(0);
            *value += 1
        }

        for (k, v) in map {
            if v == 1 {
                sum += k
            }
        }

        sum
    }
}
